### PR TITLE
feat(testing): implement comprehensive testing pyramid

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary
- Implement multi-layer testing pyramid with integration, E2E, fuzz, and mutation tests
- Add testcontainers-based integration tests for real LDAP authentication
- Add Playwright-go E2E browser tests for user journeys
- Add Go native fuzz tests for cache and web packages
- Configure mutation testing with go-mutesting for test quality validation
- Update CI workflow with new test jobs (fuzz/mutation on weekly schedule)

## Test Plan
- [x] Unit tests pass with 87% ldap_cache coverage
- [x] Integration tests build correctly (require Docker for execution)
- [x] E2E tests build correctly (require running app and Playwright)
- [x] Fuzz tests compile and seed corpus validates
- [x] Mutation testing script is executable
- [x] CI workflow syntax is valid

## New Test Infrastructure

| Layer | Location | Tests | Purpose |
|-------|----------|-------|---------|
| Integration | `internal/integration/` | ~25 | Real LDAP with testcontainers |
| E2E | `internal/e2e/` | ~28 | Browser automation with Playwright |
| Fuzz | `**/fuzz_test.go` | ~12 targets | Input validation and edge cases |
| Mutation | `scripts/mutation-test.sh` | - | Test quality assessment |

## New Makefile Targets
- `make test-integration` - Run integration tests with testcontainers
- `make test-e2e` - Run E2E browser tests
- `make test-fuzz` - Run fuzz tests (60s per target)
- `make test-mutation` - Run mutation testing
- `make test-all` - Run unit + integration tests

### CODE REVIEW

File: .github/workflows/quality.yml [hunk 1]
```diff
@@ -340,24 +340,146 @@ jobs:
...
+          pnpm install
+
+      - name: Build assets
+        run: pnpm build:assets
+
+      - name: Run fuzz tests
+        run: |
+          echo "Running fuzz tests for 60 seconds per target..."
+          go test -fuzz=FuzzCacheGetByDN -fuzztime=60s ./internal/ldap_cache/... || true
+          go test -fuzz=FuzzCacheGetBySAMAccountName -fuzztime=60s ./internal/ldap_cache/... || true
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply .github/workflows/quality.yml]: Fixed! Changed `FuzzCacheGetByDN` to `FuzzCacheFindByDN` and `FuzzCacheGetBySAMAccountName` to `FuzzCacheFindBySAMAccountName` to match the actual fuzz function names.

File: internal/web/template_errors_test.go [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+// TestTemplateCacheStopIdempotent tests that Stop can be called safely
+func TestTemplateCacheStopIdempotent(t *testing.T) {
+	cache := NewTemplateCache(DefaultTemplateCacheConfig())
+
+	// Stop should not panic on first call
+	cache.Stop()
+
+	// Note: Second Stop() would panic due to closed channel
+	// This is expected behavior - Stop should only be called once
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply internal/web/template_errors_test.go]: Fixed! Renamed the test to `TestTemplateCacheStopSafe` with clearer documentation that explains the single-call expectation.

File: scripts/mutation-test.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+NC='\033[0m' # No Color
+
+echo -e "${YELLOW}Running Mutation Testing${NC}"
+echo "========================================"
+
+# Check if go-mutesting is installed
+if ! command -v go-mutesting &> /dev/null; then
+    echo -e "${YELLOW}Installing go-mutesting...${NC}"
+    go install github.com/zimmski/go-mutesting/cmd/go-mutesting@latest
+fi
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply scripts/mutation-test.sh]: Fixed! Pinned go-mutesting to version v1.2 instead of @latest to prevent supply-chain attacks from malicious upstream changes.

File: internal/web/template_cache.go [hunk 4]
```diff
@@ -90,8 +90,8 @@ func (tc *TemplateCache) generateCacheKey(c *fiber.Ctx, additionalData ...string
 
 // Get retrieves cached template content if available and not expired
 func (tc *TemplateCache) Get(key string) ([]byte, bool) {
-	tc.mu.RLock()
-	defer tc.mu.RUnlock()
+	tc.mu.Lock()
+	defer tc.mu.Unlock()
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply internal/web/template_cache.go]: Valid point about concurrency. The Lock was used because updating accessedAt is a write operation. For this PR focused on testing infrastructure, the current implementation works correctly. Performance optimization can be a follow-up if benchmarks show it is a bottleneck.

[the remark this answers is not in the document]
Comment (CybotTM) [reply internal/web/template_cache.go]: See previous reply on this topic. The trade-off is between correctness (Lock for write to accessedAt) vs concurrency (RLock allowing multiple readers). Can be optimized in a follow-up PR if benchmarks show it is a bottleneck.

File: internal/e2e/e2e_helpers.go [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+
+	return TestConfig{
+		BaseURL:      baseURL,
+		Headless:     os.Getenv("E2E_HEADLESS") != "false",
+		SlowMo:       0,
+		Timeout:      30000,
+		AdminUser:    envOrDefault("E2E_ADMIN_USER", "admin"),
+		AdminPass:    envOrDefault("E2E_ADMIN_PASS", "adminpassword"),
+		TestUser:     envOrDefault("E2E_TEST_USER", "testuser1"),
+		TestUserPass: envOrDefault("E2E_TEST_USER_PASS", "password1"),
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply internal/e2e/e2e_helpers.go]: The E2E helpers already support environment variable overrides via the envOrDefault pattern. The defaults are for local development convenience. In CI, actual credentials are injected via environment variables.
